src/ripple/protocol/Rate.h

```diff
@@ -40,10 +41,10 @@ struct Rate
 
     Rate () = default;
 
+    explicit
     Rate (std::uint32_t rate)
```

Comment:
I wouldn't be opposed to this if we all feel that this the better approach. 

My rationale was that this should emulate a POD type because default initializing it to any one value can lead to hard to track down problem since casual inspection will show the item to have a value.

The default initialization of `STAmount` to `0` (quick: is it XRP or IOU, or maybe both?) and the default initialization of `base_uint` to `0` is, in my opinion, a problem for the same reason: it can  "obscure" issues because instances will always have that consistent initial "value".

Default initialization to some value takes away powerful tool from us: first, compilers in debug mode will "mark" uninitialized memory with a pattern (e.g. `0xCC`) that will serve as a clue if ever spotted in a debugger, and static analyzers or tools like `valgrind` can spot reads that occur before any writes.

With all that said, I could be convinced to have `Rate` default initialize to `0` if, indeed, `0` is never a valid value whenever we use `Rate`.